django/contrib/postgres/fields/array.py

```diff
@@ -70,6 +70,25 @@ def check(self, **kwargs):
                 )
         return errors
 
+    def _check_default(self):
+        if self.has_default() and self.default is not None and not callable(self.default):
+            return [
+                checks.Warning(
+                    "ArrayField uses a default instance that's shared between all field instances.",
```

Comment:
I think this method could be moved to a mixin and shared between the two classes. Using the same 'id' seems fine the the hint text could be customized with a class attribute. For the warning, I'd say something like `"%s default should use a callable instead of an instance so that it's not shared between all field instances." % self.__class__.__name__`

Reply:
Will fix this now